deepmd/pt/train/training.py

```diff
@@ -656,7 +656,8 @@ def step(_step_id, task_key="Default") -> None:
             # PyTorch Profiler
             if self.enable_profiler or self.profiling:
                 prof.step()
-            self.wrapper.train()
+            if not self.wrapper.training:
```

Comment:
I think it just checks the wrapper itself. Is there a situation that the wrapper is in the training mode but the model is in the eval mode?

Reply:
Btw, I don't think it should be put in the loop

Reply:
> I think it just checks the wrapper itself. Is there a situation that the wrapper is in the training mode but the model is in the eval mode?

It seems that the status change only happens here
https://github.com/deepmodeling/deepmd-kit/blob/53bbbdbb890e14af79fc8e8a8de778ad3cc06750/deepmd/pt/train/training.py#L769
So I guess the answer should be no.

Reply:
> Btw, I don't think it should be put in the loop

Agreed - maybe move it to the end of validation block?

Reply:
> move it to the end of validation block

Reasonable. Validation is not performed every step.